plugins/feeds/public/et_open.py

```diff
@@ -0,0 +1,138 @@
+import datetime
+import logging
+from datetime import timedelta, timezone
+from io import StringIO
+
+from idstools import rule
+
+from core import taskmanager
+from core.config.config import yeti_config
+from core.schemas import entity, indicator, task
+
+
+class ETOpen(task.FeedTask):
+    __SOURCE = (
+        "https://rules.emergingthreats.net/open/suricata-7.0.3/emerging-all.rules"
+    )
+
+    _defaults = {
+        "frequency": timedelta(days=1),
+        "name": "ETOpen",
+        "description": "ETOpen ruleset",
+        "source": __SOURCE,
+    }
+
+    def run(self):
+        response = self._make_request(self.__SOURCE, no_cache=True)
+
+        if not response:
+            return
+
+        for line in StringIO(response.text).readlines():
+            if line.startswith("#"):
+                continue
+            self.analyze(line)
+
+    def analyze(self, line: str):
+        rule_suricata = rule.parse(line)
+        if not rule_suricata:
+            return
+
+        if self.__filter_rule(rule_suricata.metadata):
```

Comment:
Using a guard statement here (if not / continue) will save you one indentation level below